source/common/http/utility.h

```diff
@@ -247,6 +247,20 @@ absl::string_view findQueryStringStart(const HeaderString& path);
  */
 std::string stripQueryString(const HeaderString& path);
 
+/**
+ * Replace the query string portion of a given path with a new one.
+ *
+ * e.g. replaceQueryString("/foo?key=1", {key:2}) -> "/foo?key=2"
```

Comment:
Since there is no standard on passing multiple values (for the same key, e.g. ?a=1&a=2, it is common that web framework perceives this as a "collection", for example: https://docs.microsoft.com/en-us/previous-versions/iis/6.0-sdk/ms524784(v=vs.90)?redirectedfrom=MSDN#remarks) in the query string, I think we need to be explicit here to do "replace all" or just the first occurrence. Or as wild as: https://github.com/ljharb/qs#parsing-arrays

Reply:
The example in the comment might be too simple and therefore misleading, but this utility function actually replaces the entire query string with the serialized representation of the given QueryParams. It doesn't attempt to parse the existing query string and do replacements on a key-by-key basis.